sol/main.c

```diff
@@ -601,37 +684,18 @@ alloc_sol_conf(void)
 
 int
 gk_solicitor_enqueue_bulk(struct sol_config *sol_conf,
-	struct rte_mbuf **pkts, uint8_t *priorities, uint16_t num_pkts)
+	struct rte_mbuf **pkts, uint16_t num_pkts)
 {
-	int i;
-	unsigned int num_enqueued;
-	struct priority_req *req_nodes[num_pkts];
-
-	if (unlikely(rte_mempool_get_bulk(sol_conf->mb.pool,
-			(void **)req_nodes, num_pkts) != 0))
-		return -1;
-
-	for (i = 0; i < num_pkts; i++) {
-		if (priorities[i] > GK_MAX_REQ_PRIORITY) {
-			SOL_LOG(WARNING, "Trying to enqueue a request with priority %hhu, but should be in range [0, %d]. Overwrite the priority to PRIORITY_REQ_MIN (%hhu)\n",
-				priorities[i], GK_MAX_REQ_PRIORITY,
-				PRIORITY_REQ_MIN);
-			priorities[i] = PRIORITY_REQ_MIN;
-		}
-
-		INIT_LIST_HEAD(&req_nodes[i]->list);
-		req_nodes[i]->pkt = pkts[i];
-		req_nodes[i]->priority = priorities[i];
-	}
-
-	num_enqueued = rte_ring_mp_enqueue_bulk(sol_conf->mb.ring,
-		(void **)req_nodes, num_pkts, NULL);
+	unsigned int num_enqueued = rte_ring_mp_enqueue_bulk(sol_conf->ring,
+		(void **)pkts, num_pkts, NULL);
 	if (unlikely(num_enqueued < num_pkts)) {
+		unsigned int i;
+
 		SOL_LOG(ERR, "Failed to enqueue a bulk of %hu requests - only %u requests are enqueued\n",
 			num_pkts, num_enqueued);
-		mb_free_entry_bulk(&sol_conf->mb,
-			(void * const *)&req_nodes[num_enqueued],
-			num_pkts - num_enqueued);
+
+		for (i = num_enqueued; i < num_pkts; i++)
+			rte_pktmbuf_free(pkts[i]);
```

Comment:
Actually, GK blocks already drop the packets and accounts for the drops. So there was no bug here. Please drop these lines. Sorry for the extra work.